components/tidb_query/src/codec/batch/lazy_column.rs

```diff
@@ -196,17 +197,17 @@ impl LazyBatchColumn {
     }
 
     /// Returns maximum encoded size.
-    pub fn maximum_encoded_size(&self, logical_rows: &[usize]) -> Result<usize> {
+    pub fn maximum_encoded_size(&self, logical_rows: &[usize]) -> usize {
         match self {
-            LazyBatchColumn::Raw(v) => Ok(v.total_len()),
+            LazyBatchColumn::Raw(v) => v.total_len(),
             LazyBatchColumn::Decoded(v) => v.maximum_encoded_size(logical_rows),
         }
     }
 
     /// Returns maximum encoded size in chunk format.
-    pub fn maximum_encoded_size_chunk(&self, logical_rows: &[usize]) -> Result<usize> {
+    pub fn maximum_encoded_size_chunk(&self, logical_rows: &[usize]) -> usize {
         match self {
-            LazyBatchColumn::Raw(v) => Ok(v.total_len()),
+            LazyBatchColumn::Raw(v) => v.total_len() * 2,
```

Comment:
Double capacity?

Reply:
@lonng  It is changed because Chunk format is very likely to be larger than Datum format (for example, see how decimal is stored). However it is very not likely that Chunk format is 2x larger than Datum format.

Reply:
@breeswish I think we should add some comments here.